weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -307,91 +319,63 @@ def calls_query_stats(self, req: tsi.CallsQueryStatsReq) -> tsi.CallsQueryStatsR
+        if req.filter is not None:
+            cq.set_hardcoded_filter(HardCodedFilter(filter=req.filter))
+        if req.query is not None:
+            cq.add_condition(req.query.expr_)
 
-        # Perform the query against the database
-        stats = self._calls_query_stats_raw(
-            req.project_id,
-            columns=list(raw_fields_used),
-            start_event_conditions=start_event_conditions,
-            end_event_conditions=end_event_conditions,
-            having_conditions=having_conditions,
-            parameters=param_builder.get_params(),
+        pb = ParamBuilder()
+        inner_query = cq.as_sql(pb)
+        raw_res = self._query(
+            f"SELECT count() FROM ({inner_query})",
+            pb.get_params(),
         )
-
-        # Return the marshaled response
-        return tsi.CallsQueryStatsRes(count=stats["count"])
+        rows = raw_res.result_rows
+        count = 0
+        if rows and len(rows) == 1 and len(rows[0]) == 1:
+            count = rows[0][0]
+        return tsi.CallsQueryStatsRes(count=count)
 
     def calls_query_stream(
         self, req: tsi.CallsQueryReq
     ) -> typing.Iterator[tsi.CallSchema]:
         """Returns a stream of calls that match the given query."""
-        having_conditions = []
-        start_event_conditions = []
-        end_event_conditions = []
-        param_builder = ParamBuilder()
-
-        # First, apply the application filter
-        if req.filter:
-            filter_to_conditions = _process_calls_filter_to_conditions(
-                req.filter, param_builder
-            )
-            having_conditions.extend(filter_to_conditions.having_conditions)
-            start_event_conditions.extend(filter_to_conditions.start_event_conditions)
-            end_event_conditions.extend(filter_to_conditions.end_event_conditions)
-
-        # Next, apply the query filter
-        if req.query:
-            having_query_conds, _ = _process_query_to_conditions(
-                req.query, all_call_select_columns, all_call_json_columns, param_builder
-            )
-            having_conditions.extend(having_query_conds)
+        cq = CallsQuery(project_id=req.project_id)
+
+        # TODO (Perf): By allowing a sub-selection of columns
+        # we will gain increased performance by not having to
+        # fetch all columns from the database. Currently all use
+        # cases call for every column to be fetched, so we have not
+        # implemented this yet.
+        columns = all_call_select_columns
+        for col in columns:
+            cq.add_field(col)
+        if req.filter is not None:
+            cq.set_hardcoded_filter(HardCodedFilter(filter=req.filter))
+        if req.query is not None:
+            cq.add_condition(req.query.expr_)
+        if req.sort_by is not None:
+            for sort_by in req.sort_by:
+                cq.add_order(sort_by.field, sort_by.direction)
+        else:
+            cq.add_order("started_at", "asc")
```

Comment:
Wondering if it's better not to have a default order if one isn't requested?

Reply:
Setting the sort field to an empty list will result in no sort. This way the default query is deterministic. Maybe we want to change that?